ui/src/pages/clearing/Member.tsx

```diff
@@ -33,22 +39,35 @@ const ClearingMemberComponent: React.FC<RouteComponentProps & ServicePageProps<S
   ({ history, services, member }) => {
     const { contractId } = useParams<any>();
     let party = useParty();
+    const { getName } = usePartyName(party);
+    const [openMarginDialog, setOpenMarginDialog] = useState(false);
+    const [deposit, setDeposit] = useState<CreateEvent<AssetDeposit>>();
+    const [observers, setObservers] = useState<AssetDeposit_SetObservers>({
+      newObservers: makeDamlSet<string>([]),
+    });
+    const [modalTitle, setModalTitle] = useState('');
 
     const ledger = useLedger();
     const service = useStreamQueries(Service).contracts.find(s =>
       !!member ? s.payload.customer === party : s.contractId === contractId
     )?.payload;
     const customer = service?.customer;
 
-    const deposits = useStreamQueries(AssetDeposit).contracts;
     const standings = useStreamQueries(MemberStanding).contracts;
-
     const standing = standings.find(standing => standing.payload.customer === customer);
+
+    const { contracts: allDeposits, loading: depositsLoading } = useStreamQueries(AssetDeposit);
+    const deposits = allDeposits.filter(
+      a => a.payload.account.id.label === service?.clearingAccount.id.label
+    );
+    const availableDeposits = deposits.filter(
+      d => isEmptySet(d.payload.lockers) && !d.payload.observers.map.has(service?.provider || '')
+    );
```

Comment:
`map.has`.. nice!

Reply:
The daml `Set` which is a daml `Map` with only keys exposes this `has` function :)